hs_core/tasks.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import absolute_import
+
+import os
+import sys
+import traceback
+import zipfile
+import logging
+import time
+
+from celery import shared_task
+
+from hs_core.hydroshare import utils
+from hs_core.models import BaseResource
+
+
+@shared_task
+def add_zip_file_contents_to_resource(pk, zip_file_path):
+    zfile = None
+    try:
+        logger = logging.getLogger('django')
+        resource = utils.get_resource_by_shortkey(pk, or_404=False)
```

Comment:
Setting or_404 to False makes sure 404 is not raised, as per [here](https://github.com/hydroshare/hydroshare/blob/178-rhessys-model-instance-phase1/hs_core/hydroshare/utils.py#L44)